runner/src/mill/runner/MillMain.scala

```diff
@@ -136,6 +177,10 @@ object MillMain {
             )
             (false, RunnerState.empty)
 
+          case Result.Success(config) if config.bsp.value != bspMode =>
```

Comment:
```suggestion
          // The first-option requirement results from sloppy parsing 
          // in MillClientMain and our executable-jar-prepend script
          case Result.Success(config) if config.bsp.value != bspMode =>
```

Reply:
That check can actually be removed, now that the actual CLI arg parser is used upfront

Reply:
Just removed it